### PR TITLE
[new release] hpack, h2, h2-mirage, h2-lwt, h2-lwt-unix and h2-async (0.9.0)

### PR DESCRIPTION
**Pull request by anmonteiro**
An HPACK (Header Compression for HTTP/2) implementation in OCaml

- Project page: <a href="https://github.com/anmonteiro/ocaml-h2">https://github.com/anmonteiro/ocaml-h2</a>
- Documentation: <a href="https://anmonteiro.github.io/ocaml-h2/">https://anmonteiro.github.io/ocaml-h2/</a>

##### CHANGES:

- h2: Fix tests on 32-bit platforms
  ([anmonteiro/ocaml-h2#152](https://github.com/anmonteiro/ocaml-h2/pull/152))
- h2-mirage: adapt to Conduit v4 and remove `H2_mirage.Server_with_conduit`
  ([anmonteiro/ocaml-h2#154](https://github.com/anmonteiro/ocaml-h2/pull/154))
- h2: fix memory leaks related to trailer headers in the server and client
  implementations ([anmonteiro/ocaml-h2#159](https://github.com/anmonteiro/ocaml-h2/pull/159)
  (thanks [@quernd](https://github.com/quernd)!),
  [anmonteiro/ocaml-h2#160](https://github.com/anmonteiro/ocaml-h2/pull/160))
- h2: allow configuring `flush_headers_immediately` on the client and default
  to `false`. This means that h2 will wait for the first request body bytes to
  be scheduled and batch `HEADERS` and `DATA` frames when sending requests
  ([anmonteiro/ocaml-h2#163](https://github.com/anmonteiro/ocaml-h2/pull/163),
  [anmonteiro/ocaml-h2#164](https://github.com/anmonteiro/ocaml-h2/pull/164))
- h2: Split `Body.t` into `Body.Writer.t` and `Body.Reader.t`
  ([anmonteiro/ocaml-h2#165](https://github.com/anmonteiro/ocaml-h2/pull/165))
- h2: OCaml 5.00 compatibility -- add `seeded_hash` to `scheduler.ml`
  ([anmonteiro/ocaml-h2#168](https://github.com/anmonteiro/ocaml-h2/pull/168))
- h2: Use a tail-recursive version of `Angstrom.skip_many`. Fixes a memory leak
  in long-running connections e.g. gRPC
  ([anmonteiro/ocaml-h2#172](https://github.com/anmonteiro/ocaml-h2/pull/172))
- h2-async: Add an OCaml-TLS client to `h2-async`
  ([anmonteiro/ocaml-h2#174](https://github.com/anmonteiro/ocaml-h2/pull/174))
- h2: Fix a bug that caused different requests to share the same headers buffer
  under concurrency ([anmonteiro/ocaml-h2#182](https://github.com/anmonteiro/ocaml-h2/pull/182))

### CODE REVIEW

**Comment by anmonteiro** [comment]
I see some revdep failures. I'll open a PR adding upper bounds to those in a bit.

**Comment by anmonteiro** [comment]
Added in https://github.com/ocaml/opam-repository/pull/21984

File: packages/hpack/hpack.0.9.0/opam [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+homepage: "https://github.com/anmonteiro/ocaml-h2"
+bug-reports: "https://github.com/anmonteiro/ocaml-h2/issues"
+dev-repo: "git+https://github.com/anmonteiro/ocaml-h2.git"
+doc: "https://anmonteiro.github.io/ocaml-h2/"
+depends: [
+  "ocaml" {>= "4.04"}
+  "dune" {>= "1.7"}
+  "yojson" {with-test}
+  "hex" {with-test}
+  "angstrom"
```
Comment (kit-ty-kate) [inline packages/hpack/hpack.0.9.0/opam:15]:
```suggestion
  "angstrom" {>= "0.7.0"}
```
```
#=== ERROR while compiling hpack.0.9.0 ========================================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.04.2 | pinned(https://github.com/anmonteiro/ocaml-h2/releases/download/0.9.0/h2-0.9.0.tbz)
# path                 ~/.opam/4.04/.opam-switch/build/hpack.0.9.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p hpack -j 31
# exit-code            1
# env-file             ~/.opam/log/hpack-7-e4ac90.env
# output-file          ~/.opam/log/hpack-7-e4ac90.out
### output ###
#       ocamlc hpack/src/.hpack.objs/byte/hpack__Decoder.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.04/bin/ocamlc.opt -w -40 -g -bin-annot -I hpack/src/.hpack.objs/byte -I /home/opam/.opam/4.04/lib/angstrom -I /home/opam/.opam/4.04/lib/cstruct -I /home/opam/.opam/4.04/lib/faraday -I /home/opam/.opam/4.04/lib/ocplib-endian -I /home/opam/.opam/4.04/lib/result -no-alias-deps -open Hpack__ -o hpack/src/.hpack.objs/byte/hpack__Decoder.cmo -c -impl hpack/src/decoder.ml)
# File "hpack/src/decoder.ml", line 68, characters 6-15:
# Error: Unbound value any_uint8
```

**Comment by kit-ty-kate** [comment]
Thanks! Could you return the fix to the hpack opam file in the repository upstream so that it can be taken into account for next time?

Also, I removed h2-async from this release as it can't be compiled. Please reopen a new PR for this particular package when it is able to be compiled.

**Pull request merged**